fix: Prevent accumulation of AppMap data while scanning

dustinbyrne opened the pull request:
Previously, event references in findings would cause entire AppMaps to stay loaded into memory. With this change we now create a shallow copy of Event objects to break these references.

I was able to confirm that the memory usage no longer grows dramatically with each AppMap scanned via the [node inspector built into Chrome](https://nodejs.org/en/docs/guides/debugging-getting-started/#inspector-clients). In my local testing, total heap size drops from ~250MB to ~35MB after a full `ci` run.

kgilpin commented:
This looks good, I think it’s worth a test case to show that the field accessors and JSON work as expected.

dustinbyrne commented:
I moved the cloning logic out into it's own file because it should really reside in @appland/models anyway. I also added some tests to validate that the cloned object performs in the same manner as the source object.

kgilpin commented on src/eventUtil.ts:
```diff
@@ -0,0 +1,50 @@
+import { CodeObject, Event } from '@appland/models';
+
+// TODO: These interfaces can be removed once @appland/models is up to date.
```
Is there an Issue captured for this?

appland-release commented:
:tada: This PR is included in version 1.40.3 :tada:

The release is available on:
- [npm package (@latest dist-tag)](https://www.npmjs.com/package/@appland/scanner/v/1.40.3)
- [GitHub release](https://github.com/applandinc/scanner/releases/tag/v1.40.3)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
